src/watchdog.rs

```diff
@@ -0,0 +1,98 @@
+use crate::hal::watchdog::{Watchdog, WatchdogEnable};
+
+use crate::stm32::{DBGMCU, IWDG};
+
+const LSI_KHZ: u32 = 40;
+const MAX_PR: u8 = 8;
+const MAX_RL: u16 = 0x1000;
+const KR_ACCESS: u16 = 0x5555;
+const KR_RELOAD: u16 = 0xAAAA;
+const KR_START: u16 = 0xCCCC;
+
+struct IndependentWatchDog {
+    iwdg: IWDG,
+}
+
+impl IndependentWatchDog {
+    pub fn new(iwdg: IWDG) -> Self {
+        IndependentWatchdog { iwdg }
+    }
+
+    pub fn stop_on_debug(&self, dbg: &DBGMCU, stop: bool) {
+        dbg.apb1_fz.modify(|_, w| w.dbg_iwdg_stop().bit(stop));
+    }
+
+    fn setup(&self, timeout_ms: u32) {
+        let mut pr = 0;
+        while pr < MAX_PR && Self::timeout_period(pr, MAX_RL) < timeout_ms {
```

Comment:
I wonder if there is a better way, using https://docs.rs/stm32f3/0.10.0/stm32f3/stm32f303/iwdg/pr/enum.PR_A.html